lib/GuestManager.php

```diff
@@ -209,4 +182,48 @@ public function getGuestInfo($userId): array {
 			}, $shares),
 		];
 	}
+
+	public function transfer(IUser $guestUser, IUser $user): void {
+		try {
+			/** @var OwnershipTransferService $ownershipTransferService */
+			$ownershipTransferService = $this->container->get(OwnershipTransferService::class);
+		} catch (QueryException $e) {
+			$this->logger->error('Could not resolve ownership transfer service to import guest user data', [
+				'exception' => $e,
+			]);
+			throw $e;
+		}
+
+		try {
+			$ownershipTransferService->transfer(
+				$guestUser,
+				$user,
+				'/',
+				null,
+				true,
+				true
+			);
```

Comment:
This could take a long time, maybe we should do it in a background job?

Reply:
Ah, I see it is copied from `Hooks.php`. I think it is good to have it triggered here instead of on the first login. But it could still take a long time, so the request could timeout.

Reply:
User feedback will need to be carefully crafted to properly reveal the state of the transfer.

Reply:
Hmm, what would you suggest UX-wise? Currently when hitting the "Transfer" button a loading spinner is shown until the server responds with a success or fail and closes the dialog

Reply:
The state would be that the new user is created, and then files will be transferred in the next cron job. After which the old user will be deleted. Maybe something like:

"The files will be transferred soon."

Maybe the @nextcloud/designers can help here :).

Reply:
Transfer is now done in a background job